lib/realtime/vehicle.ex

```diff
@@ -33,9 +49,17 @@ defmodule Realtime.Vehicle do
     :route_id,
     :trip_id,
     :current_status,
-    :stop_id
+    :stop_id,
+    :status
+  ]
+
+  @default_opts [
+    stops_on_route_fn: &Gtfs.stops_on_route/1,
+    timepoints_on_route_fn: &Gtfs.timepoints_on_route/1
   ]
 
+  @empty_timepoint %Timepoint{id: "", stop_id: ""}
```

Comment:
Setting the missing fields to `nil` instead of using default/empty data will make error handling everywhere else more explicit and less error prone.